### PR TITLE
DOC: Update python as glue doc.

### LINKED ISSUE #7035: DOC: Inaccurate caveat regarding ctypes
The "Using python as glue document" states that

> There is no standard way in Python distutils to create a standard shared library (an extension module is a “special” shared library Python understands) in a cross-platform manner. Thus, a big disadvantage of ctypes at the time of writing this book is that it is difficult to distribute in a cross-platform manner a Python extension that uses ctypes and includes your own code

But, in fact, build_ext could be used out of the box in most platforms and it's trivial to make it work in Windows. See my answer [here](http://stackoverflow.com/a/34830639/2012920). I think something like this recipe could be added to the document in place of the inaccurate and demotivating "big disadvantage" warning.

### PR DESCRIPTION
**Pull request by liang3zy22**
As @mattip suggested, to close #7035 .

### CODE REVIEW

File: doc/source/user/c-info.python-as-glue.rst
Comment (rossbar) [inline doc/source/user/c-info.python-as-glue.rst:6]:
```suggestion

.. warning::
   This was written in 2008 as part of the original
   `Guide to NumPy <https://archive.org/details/NumPyBook>`_ book
   by Travis E. Oliphant and is out of date.
```

**Review by rossbar — APPROVED** [review]
+1 for me from adding this warning. At first I was inclined to see if there was appetite to remove it entirely, but then there are sections on `f2py` which likely are still relevant for NumPy. Longer term I think it'd be worthwhile to refactor the article to focus largely on f2py and refer readers to external resource(s) for general info on extensions.

I suggest to make the note more prominent so it stands out at the top of the doc. I'll let @mattip weigh in whether there are other ways this can be improved; LGTM!

**Review by mattip — APPROVED** [review]
LGTM, thanks. Once CI runs, a reviewer should look at the rendered doc to make sure it works as expected before merging.

**Comment by rossbar** [comment]
Rendered docs look good, thanks @liang3zy22 and @mattip !

**Pull request merged**
